src/SFA.DAS.EmploymentCheck.Functions/Repositories/EmploymentCheckCacheRequestRepository.cs

```diff
@@ -24,13 +30,96 @@ public class EmploymentCheckCacheRequestRepository : IEmploymentCheckCacheReques
         public async Task Save(EmploymentCheckCacheRequest employmentCheckCacheRequest)
         {
             var dbConnection = new DbConnection();
-            await using var sqlConnection = await dbConnection.CreateSqlConnection(
+            await using (var sqlConnection = await dbConnection.CreateSqlConnection(
                 _connectionString,
-                _azureServiceTokenProvider);
-            Guard.Against.Null(sqlConnection, nameof(sqlConnection));
+                _azureServiceTokenProvider)
+            )
+            {
+                Guard.Against.Null(sqlConnection, nameof(sqlConnection));
 
+                await sqlConnection.OpenAsync();
+                using (var tran = await sqlConnection.BeginTransactionAsync())
+                {
+                    // A try/catch block is needed here to intercept any exceptions
+                    // from the database to rollback the transaction before passing
+                    // the exception up the call stack
+                    try
+                    {
+                        var existingItem = await sqlConnection.GetAsync<EmploymentCheckCacheRequest>(employmentCheckCacheRequest.Id, tran);
+                        if (existingItem != null) { await sqlConnection.UpdateAsync(employmentCheckCacheRequest, tran); }
+                        else { await sqlConnection.InsertAsync(employmentCheckCacheRequest, tran); }
+                    }
+                    catch(Exception ex)
```

Comment:
Severity	Code	Description	Project	File	Line	Suppression State
Warning	CS0168	The variable 'ex' is declared but never used	SFA.DAS.EmploymentCheck.Functions	C:\git\das\das-employmentcheck\src\SFA.DAS.EmploymentCheck.Functions\Repositories\EmploymentCheckCacheRequestRepository.cs	52	Active